python/lsst/daf/butler/tests/_dummyRegistry.py

```diff
@@ -63,11 +63,11 @@ def fetch(self, **where: Any) -> Iterator[dict]:
             if all(d[k] == v for k, v in where.items()):
                 yield d
 
-    def delete(self, **where: Any):
+    def delete(self, columns: Iterable[str], *rows: dict):
         # Docstring inherited from OpaqueTableStorage.
         kept = []
         for d in self._rows:
-            if not all(d[k] == v for k, v in where.items()):
+            if not all(d[k] == v for where in rows for k, v in where.items()):
```

Comment:
I'm not sure that I read it correctly or that it means what it is supposed to mean?

Reply:
It's correct, but I should know better than to use nested comprehensions because the ordering is very much the opposite of how it reads.  I'll rewrite it.

Reply:
But does not it also say that row should be removed from `self._rows` only when it matches _all_ rows in `rows`?

Reply:
Ah, yeah, indeed, it was completely wrong, and we got away with it because the existing test code only deleted one thing at a time because that's all it could do before.  Working on a very different and much more readable version now.